Program.cs

```diff
@@ -30,47 +27,52 @@ public static void Main(params string[] args)
                 Console.WriteLine("Ditto will still work as a local database.");
             }
 
-            Console.WriteLine("Welcome to Ditto's Task App");
+            Console.WriteLine("Welcome to Ditto's Task App\n");
+
+            string query = "SELECT * FROM tasks WHERE isDeleted == false";
+
+            ditto.Sync.RegisterSubscription(query);
 
-            subscription = ditto.Store["tasks"].Find("!isDeleted").Subscribe();
-            liveQuery = ditto.Store["tasks"].Find("!isDeleted").ObserveLocal((docs, _event) => {
-                tasks = docs.ConvertAll(document => new Task(document));
+            ditto.Store.RegisterObserver(query, (result) =>
+            {
+                tasks = result.Items.ConvertAll(item => Task.JsonToTask(item.JsonString()));
             });
 
-            ditto.Store["tasks"].Find("isDeleted == true").Evict();
+            await ditto.Store.ExecuteAsync("EVICT FROM tasks WHERE isDeleted == true");
 
             ListCommands();
 
+
             while (!isAskedToExit)
             {
-                
+
                 Console.Write("\nYour command: ");
                 string command = Console.ReadLine();
 
                 switch (command)
                 {
-                    
-                    case string s when command.StartsWith("--upsert"):
-                        string taskBody = s.Replace("--upsert ", "");
-                        ditto.Store["tasks"].Upsert(new Task(taskBody, false).ToDictionary());
+                    case string s when command.StartsWith("--insert"):
+                        string taskBody = s.Replace("--insert ", "");
+                        var task = new Task(taskBody, false).ToDictionary();
+                        await ditto.Store.ExecuteAsync("INSERT INTO tasks DOCUMENTS (:task)", new Dictionary<string, object> { { "task", task } });
                         break;
                     case string s when command.StartsWith("--toggle"):
                         string _idToToggle = s.Replace("--toggle ", "");
-                        ditto.Store["tasks"]
-                            .FindById(new DittoDocumentId(_idToToggle))
-                            .Update((mutableDoc) => {
-                                if (mutableDoc == null) return;
-                                mutableDoc["isCompleted"].Set(!mutableDoc["isCompleted"].BooleanValue);
-                            });
+                        await ditto.Store.ExecuteAsync("SELECT * FROM tasks WHERE _id == :id", new Dictionary<string, object> { { "id", _idToToggle } }).ContinueWith((result) =>
+                        {
```

Comment:
One of the reasons `async-await` was introduced is to get rid of "callback hell". So instead of doing a `ContinueWith`, the better way would've been: 

```csharp
var result = await ditto.Store.ExecuteAsync(...); 
// process the result. 
```

But this being in an ongoing while loop, I understand now it actually "blocks" the interaction. Because the UI interaction itself is not async.

Reply:
Is the idea you'd want to use either await or ContinueWith, but not both?

Reply:
Exactly. Using both is redundant. 

1. Preferred way: you'd use `await`. No need to pair it with `ContinueWith` since you get the result directly "on the next line". 
2. If the use case would not allow for await (e.g. not an async context), you could use `ContinueWith`.